lib/signs_ui_web/router.ex

```diff
@@ -29,9 +29,12 @@ defmodule SignsUiWeb.Router do
     get "/", PageController, :index
     get "/signs", SignsController, :index
     post "/signs", SignsController, :update
+    get "/messages", MessagesController, :index
   end
 
   scope "/", SignsUiWeb do
+    pipe_through [:api]
+    post "/messages/create", MessagesController, :create
```

Comment:
Specifically, from my last comment, it would be `post /messages`. The HTTP method POST is itself enough to imply that you're creating a new resource, no need to add it to the URL as well.

Reply:
That's what `resources "/messages", MessagesController, only: [:create]` would do for you here. Since we're only making one route here, I don't think we need to use that macro, but we might as well follow the conventions that it establishes.